### PR TITLE
[WIP] Ceilometer data support

### PR DESCRIPTION
**Pull request by paramite**
This is not tested yet. Please do not merge.

### CODE REVIEW

**Comment by leifmadsen** [comment]
@paramite can you provide test output showing this code working? If this code is not ready for testing, can you suggest what needs reviewing here?

**Comment by pleimer** [comment]
[https://github.com/infrawatch/smart-gateway/issues/77](url)

File: internal/pkg/events/incoming/ceilometer.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+var (
+	ceilometerAlertSeverity = map[string]string{
+		"audit":    "info",
+		"info":     "info",
+		"warn":     "warning",
+		"warning":  "warning",
+		"critical": "critical",
+		"error":    "critical",
+		"AUDIT":    "info",
```
Comment (leifmadsen) [inline internal/pkg/events/incoming/ceilometer.go]: Also need a `SAMPLE` priority type I think? I saw it in the data captured as `priority: SAMPLE`.

File: internal/pkg/events/incoming/formats.go [hunk 2]
```diff
@@ -3,6 +3,13 @@ package incoming
 import (
 	"sort"
 	"strings"
+
+	"github.com/redhat-service-assurance/smart-gateway/internal/pkg/saconfig"
```
Comment (leifmadsen) [inline internal/pkg/events/incoming/formats.go]: Needs to be `github.com/infrawatch/...` now

Comment (leifmadsen) [reply internal/pkg/events/incoming/formats.go]: Should probably merge `master` back into this patch now.

File: internal/pkg/events/incoming/formats.go [hunk 3]
```diff
@@ -75,3 +84,14 @@ func (alert *PrometheusAlert) SetSummary() {
 		}
 	}
 }
+
+//NewFromDataSource creates empty EventDataFormat accorging to given DataSource
```
Comment (leifmadsen) [inline internal/pkg/events/incoming/formats.go]: s/accorging/according

File: internal/pkg/events/incoming/formats.go [hunk 4]
```diff
@@ -75,3 +84,14 @@ func (alert *PrometheusAlert) SetSummary() {
...
+func NewFromDataSource(source saconfig.DataSource) EventDataFormat {
+	switch source {
```
Comment (leifmadsen) [inline internal/pkg/events/incoming/formats.go]: Do you need to add a `universal` type here as well?

**Comment by leifmadsen** [comment]
```
golint failed for the following reasons:

internal/pkg/events/incoming/ceilometer.go:34:6: exported type AlertKeySurrogate should have comment or be unexported
```

File: internal/pkg/events/incoming/ceilometer.go (hunk 1 above)
Comment (pleimer) [reply internal/pkg/events/incoming/ceilometer.go]: I think you are right. It does set priorities to an to `unknownSeverity` when the value is not included in the map. Maybe only the important ones are included here?

Comment (leifmadsen) [reply internal/pkg/events/incoming/ceilometer.go]: What I'm not sure about is why the priority is set to SAMPLE at all.

**Comment by leifmadsen** [comment]
@paramite please move this to the infrawatch org rather than a forked repo so that others can contribute from the team. Thanks.

**Comment by leifmadsen** [comment]
Replaced by #81

**Pull request closed without merging**